### PR TITLE
Frogger task js file

### PR DESCRIPTION
**Pull request by SergeyShytikov**
### Frogger game JS oop 

[Demo](https://sergeyshytikov.github.io/frontend-nanodegree-arcade-game/) |
[Code base](https://github.com/SergeyShytikov/frontend-nanodegree-arcade-game/tree/serhiishy_game)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by SergeyShytikov** [comment]
wow wow, keep waiting for review

**Comment by OleksiyRudenko** [comment]
> wow wow, keep waiting for review

Emmm...

> Let's do some **self-checks** to fix most common issues and to make some improvements to the code before reviewers put their hands on the code.
> 
> Go through the requirements/most common mistakes listed/linked below and fix the code as appropriate.
> 
> If you have any questions to requirements/common mistakes **feel free asking** them here or in Students' chat.
> 
> When you genuinely believe you are done **put a comment** stating that you have completed self-checks and fixed code accordingly.
> 
> Also, be aware, that if you would silently ignore this recommendation, a mentor can think that you are still working on fixes. And your PR will not be reviewed. 😒

**Comment by SergeyShytikov** [comment]
Thanks for comment, it seems I completed self-checks

File: submissions/SergeyShytikov/js-oop-frogger-task/app.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  Character.call(this, x, y, sprite);
+};
+Player.prototype = Object.create(Character.prototype);
+Player.prototype.constructor = Player;
+Player.prototype.update = function () {
+  if (this.y < TOP_BORDER) {
+    setTimeout(() => {
+      this.x = X_START_POSITION_PLAYER;
+      this.y = Y_START_POSITION_PLAYER;
+    }, "200");
```
Comment (OleksiyRudenko) [inline submissions/SergeyShytikov/js-oop-frogger-task/app.js:72]: Check if `setTimeout` expects a string literal as its second parameter.

File: submissions/SergeyShytikov/js-oop-frogger-task/app.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+};
+const allEnemies = [];
+const _allEnemies = [];
+let levelCounter = 0;
+
+function resetEnemies() {
+  if (levelCounter > 3) {
+    allEnemies.splice(0, allEnemies.length),
+      _allEnemies.splice(0, _allEnemies.length),
+      (levelCounter = 0);
```
Comment (OleksiyRudenko) [inline submissions/SergeyShytikov/js-oop-frogger-task/app.js:57]: What this fragment of code does?
Why expressions are separated with commas?

**Review by OleksiyRudenko — COMMENTED** [review]
@SergeyShytikov really good job.
But I struggle to understand a couple of fragments to provide helpful feedback.
If I was a peer developer on this project and was tasked to change something about enemies I would be puzzled.
Please answer the questions below.

File: submissions/SergeyShytikov/js-oop-frogger-task/app.js (hunk 2 above)
Comment (SergeyShytikov) [reply submissions/SergeyShytikov/js-oop-frogger-task/app.js]: comma separation is a mistake, 
and the purpose of this code is clearing the arrays with the bugs when level 3 of difficulty is reached. 
The two arrays help me to randomize the generation of the bugs more and allow them to appear on top of each other

**Comment by SergeyShytikov** [comment]
I understand that the generation of bugs is quite confusing, I will think about how to simplify it

**Comment by OleksiyRudenko** [comment]
Explain in plain language what were trying to do there.

**Comment by SergeyShytikov** [comment]
In the beginning, I assign the constants of start positions to the variables that will change during the cycle steps. I assign a random value from the range to the speed variable. then the cycle of three iterations. why three? because there are three lines along which the bugs run. in each iteration of the cycle a instance of Enemy class is created and placed in auxiliary array. then the initial speed, start position on the X-axis and the start position on the Y-axis change. And after that the auxiliary array is copied to the main enemies array. auxiliary array for more randomization. I tried to avoid the auxiliary array, but I was not satisfied with the result. it was too templated to create the bugs

**Pull request closed without merging**